sycl/include/CL/sycl/detail/pi.h

```diff
@@ -1157,12 +1157,15 @@ piextQueueGetNativeHandle(pi_queue queue, pi_native_handle *nativeHandle);
 ///
 /// \param nativeHandle is the native handle to create PI queue from.
 /// \param context is the PI context of the queue.
+/// \param device is the PI device associated with the native device used when
+///   creating the native queue. This parameter is optional but some backends
```

Comment:
We should say what device is used when it is not specified

Reply:
I don't see how we can. Some backends may be able to determine the right device from `nativeHandle`, while others (like L0) will need this to be non-null to be certain the right device is associated with the resulting `pi_queue`.

Reply:
I agree. In the end we want this to say that having `device` as `nullptr` means that the backend infers the device from the other arguments, but while we have L0 simply picking the first device in the context (which is still needed due to a seemingly old `level_zero::make_queue`) I think we have to be a little hand-wavey here. I am open to alternative wordings though!

Reply:
Please say that it is "optional" for backends that can infer the device from native queue handle.